File: statsig-php/src/EvaluationTypes/ExperimentGroupsResult.php
<?php

namespace Statsig\EvaluationTypes;

class ExperimentGroupsResult
{
    /**
     * Null when the name does not refer to an experiment (unknown name or
     * a non-experiment entity like a dynamic config or autotune);
     * otherwise the experiment's isActive state.
     */
    public ?bool $isExperimentActive;

    /** @var ExperimentGroup[] */
    public array $groups;

    public function __construct(string $raw_result)
    {
        $result = json_decode($raw_result, true);
        if (!is_array($result)) {
            $result = [];
        }

        $this->isExperimentActive = $result['is_experiment_active'] ?? null;
        $this->groups = array_map(
            fn ($group) => new ExperimentGroup($group),
            $result['groups'] ?? []
        );
    }
}

File: statsig-elixir/lib/statsig/experiment_groups_result.ex
defmodule Statsig.ExperimentGroupsResult do
  @moduledoc """
  The groups of an experiment along with its active state, exposed without
  requiring a user evaluation.

  `is_experiment_active` is `nil` when the name does not refer to an experiment
  (unknown name or a non-experiment entity like a dynamic config or autotune);
  otherwise it reflects the experiment's `isActive` state.
  """

  defstruct [
    :is_experiment_active,
    :groups
  ]

  @type t :: %__MODULE__{
          is_experiment_active: boolean() | nil,
          groups: [Statsig.ExperimentGroup.t()]
        }
end
